### PR TITLE
Add compliance pipeline for running APIScan

### PR DESCRIPTION
**Pull request by JoeRobich**
Added compliance yml for running APIScan on demand. Based on https://github.com/dotnet/project-system/blob/main/build/ci/compliance.yml

Pipeline: https://devdiv.visualstudio.com/DevDiv/_build?definitionId=16722&_a=summary (microsoft)

### CODE REVIEW

**Comment by JoeRobich** [comment]
cc: @ryanbrandenburg @NTaylorMullen @RikkiGibson

**Comment by mmitche** [comment]
@JoeRobich, @garath is looking to just enablign this for your existing Validate Build Assets stage. This may be unnecessary: https://github.com/dotnet/arcade/issues/7300

**Comment by JoeRobich** [comment]
>  is looking to just enablign this for your existing Validate Build Assets stage.

@mmitche That is great news. Looking forward to removing this.

File: azure-pipelines-compliance.yml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  timeoutInMinutes: 90
+variables:
+  BuildConfiguration: Release
+  TeamName: DotNet-Roslyn
+  SignType: test
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+  _DevDivDropAccessToken: $(System.AccessToken)
+
+steps:
+- template: eng/pipelines/checkout-windows-task.yml
```
Comment (RikkiGibson) [inline azure-pipelines-compliance.yml:22]: This file has been renamed to `checkout-task.yml`.

**Review by RikkiGibson — APPROVED** [review]
LGTM. Fine if we need to take this. If we can hold out for a solution from Arcade instead then great.

**Pull request merged**